diff --git a/bin/AnfMain.hs b/bin/AnfMain.hs
index 0fa8b05..6a2d603 100644
--- a/bin/AnfMain.hs
+++ b/bin/AnfMain.hs
@@ -19,7 +19,10 @@ runCompiler :: IO ()
 runCompiler = do
   f <- getSrcFile
   s <- readFile f
-  let out = (pprint . anormal . parse f) s
+  let af = (anormal . parse f) s
+  let out = if isAnf af
+            then "In ANF"
+            else "Not in ANF: " ++ (pprint af)
   hPutStrLn stdout out
   exitSuccess
 
diff --git a/tests/Common.hs b/tests/Common.hs
index a2bdfaf..9a79a80 100644
--- a/tests/Common.hs
+++ b/tests/Common.hs
@@ -60,7 +60,7 @@ matchSuccess expected result = trim expected == trim result
 mkAnfTest :: TestName -> Program -> Result -> IO (String, Bool)
 mkAnfTest name pgm expect = do
   res <- runHelper Anf name pgm
-  return $ check logF res expect
+  return $ check logF res (Right "In ANF")
   where
     logF = dirExt "output" name Log

